django/contrib/admin/options.py

```diff
@@ -344,8 +344,15 @@ def get_readonly_fields(self, request, obj=None):
     def get_prepopulated_fields(self, request, obj=None):
         """
         Hook for specifying custom prepopulated fields.
+
+        Initially prepopulated fields must be taken into account in any of next cases:
+            * User adds a new object.
+            * User has permission to make changes on existing object.
+        In other case it does not make sense to pass them to admin form.
```

Comment:
It also doesn't make sense to pass them to  the admin change view though, since prepopulating only works in the add form.